### PR TITLE
Fix conflict with Travis arch keyword.

### PR DESCRIPTION
**Pull request by bartoszek**
- [x] Rename `.travis.yml` current keyword from `arch` to `archlinux`
- [x] Add warning for `.travis.yml` following deprecated scheme.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ compiler:
 services:
 - docker
 
-arch:
+arch-travis:
```
Comment (mikkeloscar) [inline .travis.yml]: I like `archlinux` better, WDYT?

Comment (bartoszek) [reply .travis.yml]: Is a matter of taste - seems the same to me :smirk:
I've changed it, and incorporate a 'warning' message for the user.

File: arch-travis.sh [hunk 2]
```diff
@@ -42,11 +42,21 @@ configure_volumes() (
     while read -r vol; do printf -- '-v "%s"\n' "$vol"; done<<<"${volumes[*]}"
 )
 
+# regression test for deprecated arch-travis configuration scheme.
+{
+    if travis_yml arch script >/dev/null 2>&1; then
+      echo '*** WARNING! Your current arch-travis configuration is deprecated'
```
Comment (mikkeloscar) [inline arch-travis.sh]: This is more than deprecated as the old key won't work anymore. I think this is ok, we should just rename it to outdated or so.

Comment (bartoszek) [reply arch-travis.sh]: updated to `outdated` :smirk:

**Comment by mikkeloscar** [comment]
Thanks!

**Pull request merged**